tony-core/src/main/java/com/linkedin/tony/TaskExecutor.java

```diff
@@ -426,7 +380,91 @@ private void skewAndHangIfTesting() {
     }
   }
 
-  private boolean isGangMode() {
+  public boolean isGangMode() {
     return distributedMode == TonyConfigurationKeys.DistributedMode.GANG;
   }
+
+  public ServerPort getRpcPort() {
+    return rpcPort;
+  }
+
+  public ServerPort getTbPort() {
+    return tbPort;
+  }
+
```

Comment:
a map of ports? We have getRpcPort, getTBPort, getAmport and getMetricsRPCport here